src/ui-components/Address.tsx

```diff
@@ -56,21 +58,56 @@ const Address = ({ address, className, displayInline, disableIdenticon, extensio
 	const router = useRouter();
 	const [username, setUsername] = useState('');
 	const [kiltName, setKiltName] = useState('');
+	const[isAutoGeneratedUsername,setIsAutoGeneratedUsername] = useState(true);
 
 	const encoded_addr = address ? getEncodedAddress(address, network) || '' : '';
+	const FEATURE_RELEASE_DATE = dayjs('2023-06-12').toDate(); // Date from which web3 address is being used as username on web3 signup.
+	const manualUsernames25char: string[] = ['marcosdanieldasilvasantos','masterquincyleroyharrison','sasha.fleur.polkaassembly'];
 
-	const fetchUsername = async () => {
+	const fetchUsername = async (isOnclick:boolean) => {
 		const substrateAddress = getSubstrateAddress(address);
-		if (!username && substrateAddress) {
+
+		if (substrateAddress) {
 			const { data, error } = await nextApiClientFetch<IGetProfileWithAddressResponse>(`api/v1/auth/data/profileWithAddress?address=${substrateAddress}`);
 			if (error) {
 				console.error(error);
 				return;
 			}
-			if (data && data.username) {
-				setUsername(data.username);
-				router.push(`/user/${data.username}`);
+
+			if(isOnclick){
+				if (data && data.username) {
+					setUsername(data.username);
+					router.push(`/user/${data.username}`);
+				}
+			}else{
+
+				if(data && data.username && data.custom_username){
+					setUsername(data.username);
+					setIsAutoGeneratedUsername(false);
+					return;
+				}
+				if (data && data.username) {
+					setUsername(data.username);
+
+					if(manualUsernames25char.includes(data.username)){
+						setIsAutoGeneratedUsername(false);
+					}
+					else if(data.username.length !== 25){
+						setIsAutoGeneratedUsername(false);
+						if(data.username.length > 25){
+							setUsername(shortenAddress(data.username,8));
```

Comment:
please store full username in state instead of shortened username for back compatibilty